src/Authentication/Controllers/SystemRegisterController.cs

```diff
@@ -103,51 +105,63 @@ public async Task<ActionResult<RegisteredSystemResponse>> GetRegisteredSystemInf
+        if (attemptedUpdatedSystem.Id != systemId)
+        {
+            return BadRequest("SystemId in request body doesn't match systemId in Url");
+        }
 
-        ValidationErrorBuilder validateErrorRights = await ValidateRights(updateSystem.Rights, cancellationToken);
-        ValidationErrorBuilder validateErrorAccessPackages = await ValidateAccessPackages(updateSystem.AccessPackages, cancellationToken);
-        ValidationErrorBuilder errors = MergeValidationErrors(validateErrorRights, validateErrorAccessPackages);
+        RegisteredSystemResponse currentSystem = await _systemRegisterService.GetRegisteredSystemInfo(systemId, cancellationToken);
 
-        foreach (string clientId in updateSystem.ClientId)
+        if (currentSystem == null)
         {
-            bool clientExistsForAnotherSystem = maskinPortenClients.FindAll(x => x.ClientId == clientId && x.SystemInternalId != systemInfo.InternalId).Count > 0;
-            if (clientExistsForAnotherSystem)
-            {
-                ModelState.AddModelError("ClientId", $"ClientId {clientId} already tagged with another system");
-                return BadRequest(ModelState);
-            }
+            return NotFound($"System with ID '{systemId}' not found.");
         }
 
-        if (errors.TryToActionResult(out var errorResult))
+        List<string> allClientIds = CombineClientIds(currentSystem.ClientId, attemptedUpdatedSystem.ClientId);
+        List<MaskinPortenClientInfo> allClientIdUsages = await _systemRegisterService.GetMaskinportenClients(allClientIds, cancellationToken);
+
+        ValidationErrorBuilder validateErrorRights = await ValidateRights(attemptedUpdatedSystem.Rights, cancellationToken);
+        ValidationErrorBuilder validateErrorAccessPackages = await ValidateAccessPackages(attemptedUpdatedSystem.AccessPackages, cancellationToken);
+        ValidationErrorBuilder validateErrorClientIds = await ValidateClientIds(currentSystem, attemptedUpdatedSystem, allClientIdUsages);
+        ValidationErrorBuilder errors = MergeValidationErrors(validateErrorRights, validateErrorAccessPackages, validateErrorClientIds);
+
+        if (errors.TryToActionResult(out ActionResult errorResult))
         {
             return errorResult;
         }
 
-        var success = await _systemRegisterService.UpdateWholeRegisteredSystem(updateSystem, systemId, cancellationToken);
+        bool success = await _systemRegisterService.UpdateWholeRegisteredSystem(attemptedUpdatedSystem, systemId, cancellationToken);
 
         if (!success)
         {
-            return BadRequest();
+            return BadRequest("Unable to update system");
         }
 
+        List<MaskinPortenClientInfo> clientIdsToDelete = FindRemovedClients(currentSystem, attemptedUpdatedSystem, allClientIdUsages);
+        await DeleteRemovedClientIds(clientIdsToDelete, currentSystem.InternalId, cancellationToken);
```

Comment:
The deletion should be included in the update method within the repository class (refer to the insert client ID logic) to ensure that all related database operations are handled within a single transaction. This way, if any operation fails, the entire transaction is rolled back, which is the expected and desired behavior. We can walk through this together on a call.